### PR TITLE
[crypto] add support for verifying BLS signature shares in Move

### PR DESCRIPTION
**Pull request by alinush**
### Description

This PR exposes our BLS (multi)signature verification API inside Move smart contracts as a `bls12381_verify_signature` function.

For now, this `bls12381_verify_signature` Move API can only be (gas-efficiently) used to verify _signature shares_ that are later aggregated into a _multisignature_.

In order to (gas-efficiently) verify multisignatures, we will later add a `bls12381_aggregate_pubkey` Move API that can be used to compute an aggregate public key under which the multisignature would verify. I cannot add this right now because, AFAICT from talking to @vgao1996, Move does not support `vector<vector<u8>>` arguments in native functions. (@zekun000, @davidiw, @wrwg perhaps you two know something about this?)

Support for aggregating signature shares into multisignatures will also be added later due to the same `vector<vector<u8>>` constraint.

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2092)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/signature.move [hunk 1]
```diff
@@ -38,8 +37,25 @@ module aptos_framework::signature {
         signature: vector<u8>
     ): (vector<u8>, bool);
 
+    /// Return true if the BLS `signature` on `message` verifiers against the BLS public key `public_key`.
+    /// Returns `false` if:
+    /// - `signature` is not 96 bytes
+    /// - `public_key` is not 48 bytes
+    /// - `signature` or `public_key` are not valid: i.e., (1) they are the identity point, or (2) they are not valid
+    ///    points on the BLS12-381 elliptic curve or (3) they are not prime-order points.
+    /// - `signature` and `public key` are valid but the signature on `message` is not valid.
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/signature.move:46]: feels like we should probably add "valid" functions, abort codes, or both.

Comment (davidiw) [reply aptos-move/framework/aptos-framework/sources/signature.move:46]: well we have a validate public key

**Review by davidiw — APPROVED** [review]
I don't have a strong opinion on the aborts but ... something to think about.

File: aptos-move/framework/aptos-framework/sources/signature.move (hunk 1 above)
Comment (alinush) [reply aptos-move/framework/aptos-framework/sources/signature.move:46]: Ah, I don't know how to do this. Any examples I can look at?

Comment (alinush) [reply aptos-move/framework/aptos-framework/sources/signature.move:46]: The Ed25519 Move functions also "do not abort", so I emulated that.

**Pull request merged**